src/UCTSearch.cpp

```diff
@@ -508,8 +516,10 @@ int UCTSearch::get_best_move(passflag_t passflag) {
             // We didn't consider passing. Should we have and
             // end the game immediately?
 
+            if (!m_rootstate.is_move_legal(color, FastBoard::PASS)) {
```

Comment:
Do you mean when `bestmove` already has the value PASS? I don't think this can happen because the tree search will ignore moves that are not legal.

Reply:
I see what you are saying. I had a search through and can see that before searching from `m_root` `prepare_root_node()` calls `create_children()` and it looks like in there `PASS` is assumed to be legal and is always added to `m_children` (via `nodelist`). Maybe that should be changed instead of doing something in this function though?

Reply:
Yes. I meant in addition to what you have already done but I prefer your method of only excluding pass if there is another move that can be played